db/schema.sql

```diff
@@ -41,53 +48,65 @@ CREATE TABLE IF NOT EXISTS `ta_feedback`.`questions` (
 );
 
 CREATE TABLE IF NOT EXISTS `ta_feedback`.`dept_question_choice` (
-    `department_id` VARCHAR(50) REFERENCES `department`(`name`),
+    `department_name` VARCHAR(50),
     `term` INT NOT NULL,
-    `question_id` INT REFERENCES `questions`(`question_id`),
-    `user_id` VARCHAR(10) REFERENCES `users`(`utorid`),
+    `question_id` INT,
+    `utorid` VARCHAR(10),
     `locked` bit,
-    `position` INT NOT NULL
+    `position` INT NOT NULL,
+    FOREIGN KEY(department_name) REFERENCES department(department_name),
+    FOREIGN KEY(question_id) REFERENCES questions(question_id),
+    FOREIGN KEY(utorid) REFERENCES users(utorid)
 );
 
 CREATE TABLE IF NOT EXISTS `ta_feedback`.`course_question_choice` (
-    `survey_id` INT(0) PRIMARY KEY AUTO_INCREMENT,
-    `question_id` INT REFERENCES `questions`(`question_id`),
-    `user_id` VARCHAR(10) REFERENCES `users`(`utorid`),
+    `question_id` INT,
+    `utorid` VARCHAR(10),
     `locked` bit,
-    `position` INT NOT NULL
+    `position` INT NOT NULL,
+    FOREIGN KEY(question_id) REFERENCES questions(question_id),
+    FOREIGN KEY(utorid) REFERENCES users(utorid)
 );
 
 CREATE TABLE IF NOT EXISTS `ta_feedback`.`ta_question_choice` (
-    `survey_id` INT(0) PRIMARY KEY AUTO_INCREMENT,
     `section_id` INT,
     `term` INT NOT NULL,
-    `question_id` INT REFERENCES `questions`(`question_id`),
-    `user_id` VARCHAR(10) REFERENCES `users`(`utorid`),
+    `question_id` INT,
+    `utorid` VARCHAR(10),
     `locked` bit,
-    `position` INT NOT NULL
+    `position` INT NOT NULL,
+    FOREIGN KEY(section_id) REFERENCES sections(section_id),
+    FOREIGN KEY(question_id) REFERENCES questions(question_id),
+    FOREIGN KEY(utorid) REFERENCES users(utorid)
 );
 
 CREATE TABLE IF NOT EXISTS `ta_feedback`.`surveys` (
-    `id` INT PRIMARY KEY,
+    `survey_id` INT(0) PRIMARY KEY AUTO_INCREMENT,
     `name` VARCHAR(50) NOT NULL,
-    `course_id` VARCHAR(10) REFERENCES `courses`(`course_code`),
+    `course_code` VARCHAR(10),
     `term` INT NOT NULL,
     `default_time_window` time,
-    `default_start_time` date
+    `default_start_time` date,
+    FOREIGN KEY(course_code) REFERENCES courses(course_code)
 );
 
 CREATE TABLE IF NOT EXISTS `ta_feedback`.`survey_instances` (
-    `id` INT PRIMARY KEY,
-    `user_association_id` INT REFERENCES `user_association`(`id`),
```

Comment:
Keep this column, and remove the two below

Reply:
@david-yz-liu Thanks. I added the "user_association_id" back and removed the "section_id". Do we still need to delete the survey_instance_id? Since I notice a problem here: Each response is related to a specific survey instance by the attribute "survey_instance_id".

Reply:
@david-yz-liu Thanks!